src/server/stream_family.cc

```diff
@@ -2629,6 +2724,98 @@ void StreamFamily::XAck(CmdArgList args, ConnectionContext* cntx) {
   return;
 }
 
+void StreamFamily::XAutoClaim(CmdArgList args, ConnectionContext* cntx) {
+  ClaimOpts opts;
+  string_view key = ArgS(args, 0);
+  opts.group = ArgS(args, 1);
+  opts.consumer = ArgS(args, 2);
+  if (!absl::SimpleAtoi(ArgS(args, 3), &opts.min_idle_time)) {
+    return (*cntx)->SendError(kSyntaxErr);
+  }
+  if (opts.min_idle_time < 0) {
+    opts.min_idle_time = 0;
+  }
+
+  string_view start = ArgS(args, 4);
+  RangeId rs;
+  if (!ParseRangeId(start, &rs)) {
+    return;
+  }
+  if (rs.exclude && streamDecrID(&rs.parsed_id.val) != C_OK) {
+    return (*cntx)->SendError("invalid start ID for the interval", kSyntaxErrType);
+  }
+  opts.start = rs.parsed_id.val;
+
+  for (size_t i = 5; i < args.size(); ++i) {
+    ToUpper(&args[i]);
+    string_view arg = ArgS(args, i);
+    bool remaining_args = args.size() - i - 1 > 0;
+
+    if (remaining_args) {
+      if (arg == "COUNT") {
+        arg = ArgS(args, ++i);
+        if (!absl::SimpleAtoi(arg, &opts.count)) {
+          return (*cntx)->SendError(kInvalidIntErr);
+        }
+        if (opts.count <= 0) {
+          return (*cntx)->SendError("COUNT must be > 0");
+        }
+        continue;
+      }
+    }
+    if (arg == "JUSTID") {
+      opts.flags |= kClaimJustID;
+    } else {
+      return (*cntx)->SendError("Unknown argument given for XAUTOCLAIM command", kSyntaxErr);
+    }
+  }
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    return OpAutoClaim(t->GetOpArgs(shard), key, opts);
+  };
+  OpResult<ClaimInfo> result = cntx->transaction->ScheduleSingleHopT(std::move(cb));
+
+  if (result.status() == OpStatus::KEY_NOTFOUND) {
+    cntx->SendError(NoGroupOrKey(key, opts.group));
+    return;
+  }
+  if (!result) {
+    (*cntx)->SendError(result.status());
+    return;
+  }
+
+  ClaimInfo cresult = result.value();
+
+  (*cntx)->StartArray(3);
+  (*cntx)->SendBulkString(StreamIdRepr(cresult.end_id));
+  if (cresult.justid) {
+    (*cntx)->StartArray(cresult.ids.size());
+    for (auto id : cresult.ids) {
+      (*cntx)->SendBulkString(StreamIdRepr(id));
+    }
+  } else {
+    const RecordVec& crec = cresult.records;
+    (*cntx)->StartArray(crec.size());
+    for (const auto& item : crec) {
+      (*cntx)->StartArray(2);
+      (*cntx)->SendBulkString(StreamIdRepr(item.id));
+      (*cntx)->StartArray(item.kv_arr.size() * 2);
+      for (const auto& [k, v] : item.kv_arr) {
+        (*cntx)->SendBulkString(k);
+        (*cntx)->SendBulkString(v);
+      }
+    }
+  }
+  if (cresult.deleted_ids.size()) {
+    (*cntx)->StartArray(cresult.deleted_ids.size());
+    for (auto id : cresult.deleted_ids) {
+      (*cntx)->SendBulkString(StreamIdRepr(id));
+    }
+  } else {
+    (*cntx)->SendEmptyArray();
+  }
```

Comment:
```suggestion
  (*cntx)->StartArray(cresult.deleted_ids.size());
  for (auto id : cresult.deleted_ids) {
    (*cntx)->SendBulkString(StreamIdRepr(id));
  }
```

Reply:
(the implementation of `SendEmptyArray()` is just `SendArray(0)`, so no need for the extra code / branch)

Reply:
very nice to know, thanks!